d2/src/main/java/com/linkedin/d2/balancer/clusterfailout/ZKFailoutConfigProvider.java

```diff
@@ -100,4 +101,10 @@ public void onClusterRemoved(String clusterName)
       manager.updateFailoutConfig(null);
     }
   }
+
+  @Nullable
+  public FailedoutClusterConnectionWarmUpHandler createConnectionWarmUpHandler()
+  {
+    return null;
```

Comment:
Might want to indicate this is meant to be overridden by subclasses

Reply:
Forgot to add docstring. Fixed